src/robot/api/logger.py

```diff
@@ -66,12 +66,16 @@ def my_keyword(arg):
 """
 
 import logging
+from typing import Literal
 
 from robot.output import librarylogger
 from robot.running.context import EXECUTION_CONTEXTS
 
+LOGLEVEL = Literal['TRACE', 'DEBUG', 'INFO', 'CONSOLE', 'HTML', 'WARN', 'ERROR']
+CONSOLE_STREAM = Literal['stdout', 'stderr']
```

Comment:
- Should have two empty lines after imports.
- I believe our code handles log levels case-insensitively. This code requires them to be in upper case, but I guess that's fine.
- I also believe our code handles stream names case-insensitively. Having streams in lower case while levels are in upper cases is a bit inconsistent, but I guess that's ok.
- Both of these seem to be used only once. At least the latter one is so simple that it could just be inlined.

Reply:
I traced them back to their sources and followed the casing used there. I believe the casing for `CONSOLE_STREAM` traces back to the Python standard library and is annotated / documented as shown.

I'm pretty sure this all is indeed case-insensitive, but I don't know of a way to make the `Literal` annotation case-insensitive. In the end it boils down to whether or not the type-checkers complain about `Trace` and `trace` and not about `TRACE` while `tRacE` is actually acceptable at runtime.

Reply:
> * Should have two empty lines after imports.

Done in the different modules

> * I believe our code handles log levels case-insensitively. This code requires them to be in upper case, but I guess that's fine.
> * I also believe our code handles stream names case-insensitively. Having streams in lower case while levels are in upper cases is a bit inconsistent, but I guess that's ok.

Handling is indeed case-insensitive. I haven't found a way to have case-insensitive annotations for Literals, so following the casing as used in the Python docs is the preferred option in my option.

> * Both of these seem to be used only once. At least the latter one is so simple that it could just be inlined.

Inlined means a line break (more than 88 characters) but I guess that's preferred to the `CONSOLE_STREAM` alias that, in hindsight is not a really good name since it's actually referring to `sys.stdout` and `sys.stderr` (not really related to console, essentially).